### ISSUE TITLE
"C_Cpp.clang_format_path" option has no effect

### ISSUE THREAD

**Issue by harmishhk** [microsoft/vscode-cpptools#197]
I have noticed that in update 0.9.0 of the C++ extension formatting related settings have been moved from .vscode/c_cpp_properties.json to user setting (settings.json), which was hard to find as it is not yet documented.

Anyway, in setting.json the intellisense suggest "C_Cpp.clang_format_path" option to set path to clang_format executable, which has no effect. Instead I need to set clang_format path with "c_cpp.clang_format_path" (note the c_cpp in lower case) for formatting with Ctrl+Shift+I to work.

**Comment by sean-mcmanus** [comment]
Yeah, sorry about the lack of documentation on the breaking settings change.

I'm not able to repro the behavior you describe on Windows/Linux. Setting "C_Cpp.clang_format_path" works for me and the lowercase version doesn't work.

**Comment by sridmad** [comment]
@sean-mcmanus let's document the changes in readme under what's changed section

**Comment by harmishhk** [comment]
After digging a little deeper I found that, in fact, the `"C_Cpp.clang_format_path"` option works correctly, however only with `clang-format` version >=3.8.0. I was providing a path to `clang-format` executable version 3.6.0 (which is the default version for ubuntu 14.04). In my case, when I wrote `c_cpp` in lower case the extension was just ignoring this option and using the default value of `"C_Cpp.clang_format_path"` option. 

I also found that now the c++ extension comes with (or it downloaded) `clang-format` version 3.8.1 in the `~/.vscode/extensions/ms-vscode.cpptools-0.9.0/LLVM/bin` directory, and it was using this version when `"C_Cpp.clang_format_path"` option is left unchanged.

So, the issue is no more relevant. Closing it.
